pkg/kedge/http/lbtransport/body_test.go

```diff
@@ -0,0 +1,96 @@
+package lbtransport
+
+import (
+	"bytes"
+	"io"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestLazyBufferedReader(t *testing.T) {
+	for _, tcase := range []struct {
```

Comment:
Given the number of test cases I'd suggest adding names to each of them that can be used to run the corresponding sub-test. It also allows people to easily check whether a given edge-cases they've come up with is actually tested for.